sdk/appconfiguration/spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ClientStore.java

```diff
@@ -0,0 +1,172 @@
+        }
+
+        String clientId = Optional.ofNullable(connection)
+            .map(Connection::getClientId)
+            .orElse(null);
+        boolean clientIdIsPresent = StringUtils.isNotEmpty(clientId);
+        boolean tokenCredentialIsPresent = tokenCredential != null;
+        boolean connectionStringIsPresent = Optional.ofNullable(connection)
+            .map(Connection::getConnectionString)
+            .filter(StringUtils::isNotEmpty)
+            .isPresent();
+        boolean endPointIsPresent = Optional.ofNullable(connection)
+            .map(Connection::getEndpoint)
+            .filter(StringUtils::isNotEmpty)
+            .isPresent();
+        if ((tokenCredentialIsPresent || clientIdIsPresent)
+            && connectionStringIsPresent
+        ) {
+            throw new IllegalArgumentException(
+                "More than 1 Conncetion method was set for connecting to App Configuration.");
+        } else if (tokenCredential != null && clientIdIsPresent) {
+            throw new IllegalArgumentException(
+                "More than 1 Conncetion method was set for connecting to App Configuration.");
+        }
+
+        if (tokenCredential != null) {
+            // User Provided Token Credential
+            LOGGER.debug("Connecting to " + endpoint + " using AppConfigurationCredentialProvider.");
+            builder.credential(tokenCredential);
+        } else if (clientIdIsPresent && endPointIsPresent) {
+            // User Assigned Identity - Client ID through configuration file.
+            LOGGER.debug("Connecting to " + endpoint + " using Client ID from configuration file.");
+            ManagedIdentityCredentialBuilder micBuilder = new ManagedIdentityCredentialBuilder()
+                .clientId(clientId);
+            builder.credential(micBuilder.build());
+        } else if (connectionStringIsPresent) {
+            // Connection String
+            LOGGER.debug("Connecting to " + endpoint + " using Connecting String.");
+            builder.connectionString(connection.getConnectionString());
+        } else if (endPointIsPresent) {
+            // System Assigned Identity. Needs to be checked last as all of the above
+            // should have a Endpoint.
+            LOGGER.debug("Connecting to " + endpoint
+                + " using Azure System Assigned Identity or Azure User Assigned Identity.");
+            ManagedIdentityCredentialBuilder micBuilder = new ManagedIdentityCredentialBuilder();
+            builder.credential(micBuilder.build());
+        } else {
+            throw new IllegalArgumentException("No Configuration method was set for connecting to App Configuration");
+        }
+
+        builder.endpoint(endpoint);
+
+        if (clientProvider != null) {
+            clientProvider.setup(builder, endpoint);
+        }
+
+        builder.httpLogOptions(new HttpLogOptions().setApplicationId(AZURE_SPRING_CLOUD_APP_CONFIGURATION));
```

Comment:
@saragluna , @mrm9084, FYI:

I set applicationId here to track the usage count which will be used in BI.